### PR TITLE
Added handling PlatformStatus.Type other than Baremetal

### PR DESCRIPTION
**Pull request by jacob-anders**

### CODE REVIEW

File: provisioning/utils.go [hunk 1]
```diff
@@ -48,21 +49,30 @@ func getPodHostIP(podClient coreclientv1.PodsGetter, targetNamespace string) (st
 	return hostIP, err
 }
 
-func GetServerInternalIP(osclient osclientset.Interface) (string, error) {
+func GetServerInternalIP(osclient osclientset.Interface, config *metal3iov1alpha1.ProvisioningSpec, client kubernetes.Interface, targetNamespace string) (string, error) {
```
Comment (dtantsur) [inline provisioning/utils.go]: Let's keep the argument order the same as GetIronicIP

File: provisioning/utils.go [hunk 2]
```diff
@@ -48,21 +49,30 @@ func getPodHostIP(podClient coreclientv1.PodsGetter, targetNamespace string) (st
...
-		err = fmt.Errorf("Cannot detect server API VIP: not a baremetal platform")
+	switch infra.Status.PlatformStatus.Type {
+	case osconfigv1.BareMetalPlatformType, osconfigv1.OpenStackPlatformType, osconfigv1.VSpherePlatformType:
+		// FIXME(dtantsur): handle the new APIServerInternalIPs field and the dualstack case.
+		return infra.Status.PlatformStatus.BareMetal.APIServerInternalIP, nil
+	//NOTE(janders) how about osconfigv1.OvirtPlatformStatus, osconfigv1.KubevirtPlatformStatus, osconfigv1.NutanixPlatformStatus?
+	// they do have the APIServerInternalIP field but I suppose they don't support disabling provisioning network?
+	case osconfigv1.NonePlatformType:
+		//TODO(janders) do things the old way, reusing master code to get the patch up, further discussion needed
+		//(temporarily) re-added all the parameters needed by this code
```
Comment (dtantsur) [inline provisioning/utils.go]: Yeah.. maybe we return `"", nil` from here and use this as a sign to fall-back in `GetIronicIP`?

Comment (dtantsur) [reply provisioning/utils.go]: Although... it makes a confusing public API for this function, so maybe we want to take it private...

File: provisioning/utils.go [hunk 3]
```diff
@@ -48,21 +49,30 @@ func getPodHostIP(podClient coreclientv1.PodsGetter, targetNamespace string) (st
...
 		return "", err
 	}
-
-	// NOTE(dtantsur): do we need to handle non-BareMetal platforms here?
-	if infra.Status.PlatformStatus.BareMetal == nil {
-		err = fmt.Errorf("Cannot detect server API VIP: not a baremetal platform")
+	switch infra.Status.PlatformStatus.Type {
+	case osconfigv1.BareMetalPlatformType, osconfigv1.OpenStackPlatformType, osconfigv1.VSpherePlatformType:
+		// FIXME(dtantsur): handle the new APIServerInternalIPs field and the dualstack case.
+		return infra.Status.PlatformStatus.BareMetal.APIServerInternalIP, nil
```
Comment (dtantsur) [inline provisioning/utils.go:61]: The `BareMetal` part will be different for OpenStack and VSphere (yay for copy-pasting in Go)

File: provisioning/utils.go [hunk 4]
```diff
@@ -48,21 +49,30 @@ func getPodHostIP(podClient coreclientv1.PodsGetter, targetNamespace string) (st
...
+	// they do have the APIServerInternalIP field but I suppose they don't support disabling provisioning network?
+	case osconfigv1.NonePlatformType:
+		//TODO(janders) do things the old way, reusing master code to get the patch up, further discussion needed
+		//(temporarily) re-added all the parameters needed by this code
+		if config.ProvisioningNetwork != metal3iov1alpha1.ProvisioningNetworkDisabled && !config.VirtualMediaViaExternalNetwork {
+			return config.ProvisioningIP, nil
+		} else {
+			return getPodHostIP(client.CoreV1(), targetNamespace)
+		}
+	default:
```
Comment (dtantsur) [inline provisioning/utils.go:66]: Let's leave a comment in https://github.com/openshift/cluster-baremetal-operator/blob/master/controllers/provisioning_enabled.go#L43 to update this code whenever that part is updated.

File: provisioning/utils.go (hunk 1 above)
Comment (jacob-anders) [reply provisioning/utils.go]: Good point. Done!

File: provisioning/utils.go (hunk 4 above)
Comment (jacob-anders) [reply provisioning/utils.go:66]: Good point, done.

File: provisioning/utils.go (hunk 3 above)
Comment (jacob-anders) [reply provisioning/utils.go:61]: Well spotted and my bad... fixed now, thank you!

File: provisioning/utils.go (hunk 2 above)
Comment (jacob-anders) [reply provisioning/utils.go]: Thank you for the pointers, all done.

**Comment by dtantsur** [comment]
Time to propose this against openshift

**Comment by jacob-anders** [comment]
ACK, will do, thank you @dtantsur

**Pull request closed without merging**